lib/api/model/events.dart

```diff
@@ -363,6 +369,75 @@ enum MessageType {
   private;
 }
 
+/// A Zulip event of type `reaction`.
+abstract class ReactionEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'reaction';
+
+  String get op;
+
+  ReactionEvent({required super.id});
+}
```

Comment:
All of `ReactionAddEvent`'s and `ReactionRemoveEvent`'s fields are the same except `op`… do you think it would be better to deduplicate by putting the fields on this `ReactionEvent` base class?

Reply:
Yeah, I think so. I think the agreement in list of fields is not really a coincidence: this reactions data model is designed so that all these fields give the coordinates of one single hypothetical reaction, and then the only things that can happen are that that hypothetical reaction is created or deleted.

Potentially even go further: just have `ReactionEvent`, make it concrete, and give it `final ReactionOp op;` where that's an enum with values add/remove.

Relatedly, in the consuming code, I think we don't care whether it's an add or a remove until relatively late in handling it. For example `MessageListView` can sensibly have just one method that takes a `ReactionEvent` and does the appropriate thing (which means `PerAccountStore.handleEvent` doesn't care about the details), and then that method can go use `messageId` to find the relevant message (and potentially determine it doesn't have it and there's nothing to do) and only after finding the message does it consult `op`.